### ISSUE TITLE
Overload of memory after last iteration of fit() method

### ISSUE THREAD

**Issue by DavidBert** [catboost/catboost#357]
Hello,
I'm currently having issues by calling the fit() method.
I'm training a CatboostRegressor on a (1202739, 47) train set with 8 categorical features and with a (300685, 47) validation set.
I'm running the regressor with a depth of 7 and with 4000 iterations
Fitting actually goes well until last iteration and I even get the best_iter message.
But then it seems that RAM increases drastically and freezes my computer.
Can you tell me why?

I'm running Catboost on CPU on Ubuntu 16GB of ram  and 16GB of swap and using Jupyter
I actually got the same issue using Google Colaboratory
Catboost version 0.8.1.1

Many thanks,

David

**Comment by kizill** [comment]
@DavidBert , this looks like model CTR tables calculation, we currently don't limit final CTR calculation parallelism, it's bug. How many CPU cores do you have?

**Comment by DavidBert** [comment]
I got 4

**Comment by DavidBert** [comment]
@kizill Is there any way to avoid this issue?

**Comment by thomasht86** [comment]
Experiencing same issue with Catboost version 0.8.1 on Ubuntu 16.04 with 64GB RAM and 16 cores. Training dataset with ~2.7M rows and 1500 columns. 
Are there any workarounds? What would be the best way to limit CTR calculations without hurting performance much?

**Comment by andrey-khropov** [comment]
We're working on a solution and expecting it to be merged into master early next week.

> Are there any workarounds? What would be the best way to limit CTR calculations without hurting performance much?

You can try to decrease 'max_ctr_complexity' parameter and also decrease the number of iterations. This will help to reduce memory usage on the last stage of the training, but the trained model could have slightly worse performance. But this should be no longer needed after the upcoming fix.

**Comment by annaveronika** [comment]
the fix is on github now
